src/pages/VenuePage/VenuePage.tsx

```diff
@@ -173,20 +263,44 @@ export const VenuePage: React.FC = () => {
 
   // useVenueAccess(venue, handleAccessDenied);
 
-  if (venueRequestStatus && !venue) {
+  if (isVenueLoaded && !venue) {
     return <>This venue does not exist</>;
   }
 
-  if (!venue || !venueId) {
-    return <LoadingPage />;
+  if (!venueId) {
+    // this should be happening only if invalid url param
+    // no sense in displaying loading page, another message or action is needed
+    // @debt use/display proper message/action when !venueId
+    return null;
+  }
```

Comment:
Makes sense, but this shouldn't be `@debt`, it should be done in this PR, if even only as a hacky 1 liner 'black screen error'

Reply:
Asked product for proper message, can always duplicate the original Loading... if we need something quick.

Reply:
OOS for performance reasons, UX issue to be resolved in latter PR. Reverting to original behavior in 6e1ca9a6086f3a008b5e180b5c0f676ffc041046 with added comment
>// if !venueID is true loading page might display indefinitely, another message or action may be appropriate

Reply:
@goran-peoski-work Can you please prefix that comment with `@debt`

Reply:
done https://github.com/sparkletown/sparkle/pull/1713/commits/923a6499ed118c310a50b27bd029f207cdad8701

Reply:
done https://github.com/sparkletown/sparkle/pull/1713/commits/923a6499ed118c310a50b27bd029f207cdad8701